src/main/java/seedu/address/model/person/Person.java

```diff
@@ -74,6 +75,25 @@ public Set<Group> getGroups() {
         return Collections.unmodifiableSet(groups);
     }
 
+    /**
+     * Retrieves the first {@link Group} from the set of groups.
+     *
+     * <p>This method returns the first element of the unmodifiable set of groups
+     * if available. If the set is empty, it returns {@code null}.</p>
+     *
+     * @return the first {@link Group} if it exists, or {@code null} if the set
+     *         of groups is empty.
+     *
+     */
+    public Group getGroup() {
+        Iterator<Group> iterator = getGroups().iterator();
```

Comment:
Is there a particular reason for using an iterator instead of a generic list? I believe an iterator may introduce more overhead as it has to maintain the current state internally of the current position.  Every call to next() in the iterator adds an additional function call overhead.